### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by sirdeggen** [bitcoin-sv/ts-sdk#31]
## Bug Description

Seems like you can create a PrivateKey which is out of bounds with respect to the finite field. Also Wif encoding is bad for small privateKeys, I assume something is not getting padded or something?

## Steps to Reproduce

```javascript
import { PrivateKey, PublicKey, Curve, Hash, BigNumber } from '@bsv/sdk';
const c = new Curve()
const G = c.g
const bn = new BigNumber(c.n + 12)
const sn = new BigNumber(12)
const o = PrivateKey.fromString(bn.toHex(), 'hex')
const os = PrivateKey.fromString(sn.toHex(), 'hex')
console.log({ oWif: o.toWif(), oHex: o.toHex(), mul: G.mul(bn).encode(true, 'hex') })
console.log({ osWif: os.toWif(), osHex: os.toHex(), muls: G.mul(sn).encode(true, 'hex') })
```

## Expected Behavior

Error on the first one, because the PrivateKey is out of bounds, it needs to be mod N.

The actual Wif should be: KwDiBf89QgGbjEhKnhXJuH7LrciVrZi3qYjgd9M7rFU79MFFcB1G  but the publicKey in hex is correct.

```javascript
Error: PrivateKey Big Number is out of range (N)
{
  osWif: 'KwDiBf89QgGbjEhKnhXJuH7LrciVrZi3qYjgd9M7rFU79MFFcB1G',
  osHex: 'c',
  muls: '03d01115d548e7561b15c38f004d734633687cf4419620095bc5b0f47070afe85a'
}
```

## Actual Behavior

```javascript
{
  oWif: '2Sq75qDQa51hU2sCJk1SFKMVrrTUCDSuQxDR5toePt1z5q3yrzP3Hy',
  oHex: '63ffffffffffffffffffffffffffffff80ec4e4a1c785e9756ee2cef0155317d70',
  mul: '03d01115d548e7561b15c38f004d734633687cf4419620095bc5b0f47070afe85a'
}
{
  osWif: '5rSV7nEaL3',
  osHex: 'c',
  muls: '03d01115d548e7561b15c38f004d734633687cf4419620095bc5b0f47070afe85a'
}
```

## Environment

- `@bsv/sdk` version: 1.0.9

## Additional Information

Edge cases, but it is important too within the context of deriveChild logic. We should double check that we're not using out of bounds Private Keys now and then because we're not modding the hmac by N before use. I think it still works out to the same number, but might be worth double checking.

**Comment by tonesnotes** [comment]
Changes to resolve these issues:

Default PrivateKey constructor now forces its input value, however provided, to the curve by applying mod N.

toWif method now forces binary value to 32 BE bytes before base 58 encoding.

Additional details:

The PrivateKey constructor now has an additional optional argument modN which controls the "in curve field" limiting behavior.

The options are:

'apply': the default, apply modN to input to guarantee a valid PrivateKey.
'error': if input is out of field throw Error('Input is out of field').
'nocheck': assumes input is in field or will become so prior to use as a private key.
The toWif method checks that the BigNumber value is in field and throws an error if not.

**Comment by tonesnotes** [comment]
Waiting for review prior to merge...

**Comment by tonesnotes** [comment]
https://github.com/bitcoin-sv/ts-sdk/pull/33

**Comment by ty-everett** [comment]
Should be fixed by #33
